examples/detailed_example.py

```diff
@@ -0,0 +1,96 @@
+import json
+import logging
+import os
+
+import aioredis
+
+import sanic
+from sanic import Sanic
+
+
+ENV_VARS = ["REDIS_HOST", "REDIS_PORT",
+            "REDIS_MINPOOL", "REDIS_MAXPOOL",
+            "REDIS_PASS", "APP_LOGFILE"]
+
+app = Sanic(name=__name__)
+
+logger = None
+
+
+@app.middleware("request")
+async def attach_db_connectors(request):
+    if not hasattr(request.app.config, "REDIS"):
+        logger.info("Setting up connection to Redis Cache")
+        request.app.config.REDIS = await aioredis.create_pool((app.config.REDIS_HOST, int(app.config.REDIS_PORT)),
```

Comment:
After going back to the code, reading the [docs](https://sanic.readthedocs.io/en/latest/sanic/request_data.html) and running some tests, the connection pool is only being made once. Technically, request.app is a reference to the global app that is created.  It is then stored in app.config.REDIS to be reused (pull a connection from the pool) for future requests. 

This could probably be documented a bit better in my example to showcase that. Originally I was going to make this example multiple files. Not sure if that is the best way, but I can change it to be that if it would make this point clearer.